CODE_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,73 @@
+# Code Review Check List
+
+The goal of the code review checklist is to start productive questions
+during code review to help ensure that we are constantly improving the
+quality of the Chef Server.
+
+Change authors should review the checklist before submitting their
+pull request. Reviewers should consult the checklist during review in
+addition to reviewing the code for any language-specific standards we
+may have.
+
+- [ ] Do the commit messages and PR description give enough context on why
+      this change is being made? Would someone reading the commit message
+      6 months from now find it useful?
+
+- [ ] Does the code work? Does it compile without warnings?
+
+- [ ] Are the travis-based unit tests and the pedant test passing?
+
+- [ ] Are there integration tests for new user-facing features?
+
+- [ ] If this is a bug fix, is there a regression test for the bug?
+
+- [ ] Are the included tests likely to prevent future breakage of this
+      feature?
+
+- [ ] Has this PR changed the format of any on disk files that may be
+      used by add-ons? If so, has add-on compatibility been
+      specifically tested?
+
+- [ ] Has this PR changed an API response that may break API clients?
```

Comment:
If we care about API clients (haha, of course we do), would it make sense to enumerate those _we support_? As in, you can't fix the world's Chef server API clients, but we can take care of chef-client. (_Or is_ {chef-client} that set?)

Also, we're not mentioning API versioning **or** versioning altogether -- would it make sense to add some item for that? I'd assume the contributed change itself won't bump the version, but **the necessity to bump the version** might be relevant for code review?